tests/virtualization/universal/windows.pm

```diff
@@ -27,13 +27,20 @@ sub remove_guest {
 sub run {
     my $self = shift;
     my $username = 'Administrator';
+    my $win2k19_mac = $virt_autotest::common::imports{win2k19}->{macaddress};
 
     # Remove already existing guests to ensure a fresh start (needed for restarting jobs)
     remove_guest $_ foreach (keys %virt_autotest::common::imports);
     #    shutdown_guests();    # Shutdown SLES guests as they are not needed here
 
     import_guest $_, 'virt-install' foreach (values %virt_autotest::common::imports);
-    add_guest_to_hosts $_, $virt_autotest::common::imports{$_}->{ip} foreach (keys %virt_autotest::common::imports);
+
+    # Wait for win2k19 boot and dhcp request, and get the IP address using nmap
+    sleep 60;
+    my $win2k19_ip = script_output_retry("nmap -sn 192.168.122.0/24 | grep $win2k19_mac -B2 | head -1 | grep -oE '[0-9]+.[0-9]+.[0-9]+.[0-9]+'", delay => 10, retry => 10);
+
+    # Add the guest to hosts
+    add_guest_to_hosts $_, $win2k19_ip foreach (keys %virt_autotest::common::imports);
```

Comment:
If we'd like to keep it good for `foreach (keys %virt_autotest::common::imports)`, I think we shall move above mac query and ip query based on mac , together to be within the `foreach (keys %virt_autotest::common::imports)`.

Reply:
To be honest I was wondering why foreach was used when we only test 1 guest, but I can rework it to use the foreach for the mac/ip finding. It can future proof us for testing multiple win2k* guests.

Reply:
Indeed, that's the point -- we will have high chance for multiple windows guests in test.

Reply:
@alice-suse 
I have made the changes. Here are two verification runs:
http://openqa.qam.suse.cz/tests/74464#dependencies
http://openqa.qam.suse.cz/tests/74490#dependencies